geopandas/geodataframe.py

```diff
@@ -225,6 +277,33 @@ def rename_geometry(self, col, inplace=False):
         self.rename(columns={geometry_col: col}, inplace=inplace)
         self.set_geometry(col, inplace=inplace)
 
+    @property
+    def crs(self):
+        """
+        The Coordinate Reference System (CRS) represented as a ``pyproj.CRS``
+        object.
+
+        Returns None if the CRS is not set, and to set the value it
+        :getter: Returns a ``pyproj.CRS`` or None. When setting, the value
+        can be anything accepted by
+        :meth:`pyproj.CRS.from_user_input() <pyproj.crs.CRS.from_user_input>`,
+        such as an authority string (eg "EPSG:4326") or a WKT string.
+        """
+        return self._crs
+
+    @crs.setter
+    def crs(self, value):
+        """Sets the value of the crs"""
+        if self._geometry_column_name not in self:
+            self._crs = None if not value else CRS.from_user_input(value)
```

Comment:
Did you encounter in the tests? I would say, if there is no geometry column, should we raise when trying to set the CRS?

Reply:
Yes. In case of GeoDataFrame, you can have one without geometry but with CRS and then pass geometry. If that comes without CRS, it will inherit df's one. For that reason it is stored on gdf level as well (which also answers the question above I guess).